src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,731 @@
+        AllocationDelayInfo memory info = _allocationDelayInfo[operator];
+
+        if (info.pendingDelay != 0 && block.timestamp >= info.pendingDelayEffectTimestamp) {
+            return (true, info.pendingDelay);
+        } else {
+            return (info.delay != 0, info.delay);
+        }
+    }
+
+    /**
+     * @param operator the operator to get the slashable magnitude for
+     * @param strategies the strategies to get the slashable magnitude for
+     *
+     * @return operatorSets the operator sets the operator is a member of and the current slashable magnitudes for each strategy
+     */
+    function getSlashableMagnitudes(
+        address operator,
+        IStrategy[] calldata strategies
+    ) external view returns (OperatorSet[] memory, uint64[][] memory) {
+        OperatorSet[] memory operatorSets = avsDirectory.getOperatorSetsOfOperator(operator, 0, type(uint256).max);
+        uint64[][] memory slashableMagnitudes = new uint64[][](strategies.length);
+        for (uint256 i = 0; i < strategies.length; ++i) {
+            slashableMagnitudes[i] = new uint64[](operatorSets.length);
+            for (uint256 j = 0; j < operatorSets.length; ++j) {
+                slashableMagnitudes[i][j] = uint64(
+                    _magnitudeUpdate[operator][strategies[i]][_encodeOperatorSet(operatorSets[j])].upperLookupLinear(
+                        uint32(block.timestamp)
+                    )
+                );
+            }
+        }
+        return (operatorSets, slashableMagnitudes);
+    }
+
+    /**
+     * @notice Get the allocatable magnitude for an operator and strategy based on number of pending deallocations
+     * that could be completed at the same time. This is the sum of freeMagnitude and the sum of all pending completable deallocations.
+     * @param operator the operator to get the allocatable magnitude for
+     * @param strategy the strategy to get the allocatable magnitude for
+     */
+    function getAllocatableMagnitude(address operator, IStrategy strategy) external view returns (uint64) {
+        OperatorMagnitudeInfo storage info = operatorMagnitudeInfo[operator][strategy];
+        (uint64 freeMagnitudeToAdd,) = _getPendingFreeMagnitude({
+            operator: operator,
+            strategy: strategy,
+            numToComplete: type(uint16).max,
+            nextIndex: info.nextPendingFreeMagnitudeIndex
+        });
+        return info.freeMagnitude + freeMagnitudeToAdd;
+    }
+
+    /// @notice operator is slashable by operatorSet if currently registered OR last deregistered within 21 days
+    function isOperatorSlashable(address operator, OperatorSet memory operatorSet) public view returns (bool) {
+        bool isMember = avsDirectory.isMember(operator, operatorSet);
+        if (isMember) {
+            return true;
+        }
+        (, uint32 lastDeregisteredTimestamp) =
+            avsDirectory.operatorSetStatus(operatorSet.avs, operator, operatorSet.operatorSetId);
```

Comment:
Is it necessary to have 2 external calls here?

Can we just call `avsDirectory.operatorSetStatus` and look at both return values?

Reply:
(if possible we should explore these being the same contract -- we'll see how the refactor goes and if that's possible again)

Reply:
Fixed in: #786

Solution: Moved `isOperatorSlashable` -> AVSD since all relevant storage lives there.